src/framework/multiinstances/internal/multiinstancesprovider.cpp

```diff
@@ -231,7 +234,21 @@ void MultiInstancesProvider::activateWindowWithoutProject()
     mainWindow()->requestShowOnBack();
 #endif
 
-    m_ipcChannel->broadcast(METHOD_ACTIVATE_WINDOW_WITHOUT_PROJECT, {});
+    ID idWithNoProject;
+    m_ipcChannel->syncRequestToAll(METHOD_IS_WITHOUT_PROJECT, {}, [&idWithNoProject](const QStringList& retArgs, const ID& srcId) {
+        IF_ASSERT_FAILED(!retArgs.empty()) {
+            return false;
+        }
+        if (retArgs.at(0).toInt()) {
+            idWithNoProject = srcId;
```

Comment:
slightly improvement might be to always capture the lowest-valued ID here, so behaviour is a bit more consistent